### PR TITLE
Create test

### LINKED ISSUE #7: Flashcardservice: testing
## Description
Currently there are no test in the flashcardservice. These need to be added.
Also there could be done some refactoring.

## Acceptance Criteria
- [ ] Unit tests exist and work
- [ ] Integration tests exist and work

### PR DESCRIPTION
**Pull request by Keyshal-211**
## Description of changes
Please provide a short description of the change or if the change is sufficiently described in the GitHub issue, provide a link to the issue.

  
## How has this been tested?
Please describe the test strategy you followed.

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [ ] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.
In case of automated tests, the test coverage should be reasonably high, especially on Service classes and other classes with complex logic.
    
## Checklist before requesting a review
- [X] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My code fulfills all acceptance criteria
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [X] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: src/main/java/de/unistuttgart/iste/gits/flashcard_service/persistence/repository/FlashcardSideRepository.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package de.unistuttgart.iste.gits.flashcard_service.persistence.repository;
+
+
+import de.unistuttgart.iste.gits.flashcard_service.persistence.dao.FlashcardSideEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+
+import java.util.UUID;
+
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcard_service/persistence/repository/FlashcardSideRepository.java]: Add the `@Repository` annotation here #NitPick

File: src/test/java/de/unistuttgart/iste/gits/flashcard_service/api/mutation/MutationCreateFlashcardSetTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+@TablesToDelete({"flashcard_side", "flashcard", "flashcard_set"})
+class MutationCreateFlashcardSetTest {
+    @Autowired
+    private FlashcardRepository flashcardRepository;
+    @Test
+    @Transactional
+    @Commit
+    void testCreateFlashcardSet(GraphQlTester graphQlTester) {
+        UUID assessmentId = UUID.randomUUID();
+        String query = """
```
Comment (PaulBredl) [inline src/test/java/de/unistuttgart/iste/gits/flashcard_service/api/mutation/MutationCreateFlashcardSetTest.java:28]: Please format this query in a better way, the formatting is all over the place

File: src/test/java/de/unistuttgart/iste/gits/flashcard_service/api/mutation/query/QueryByAssessmentIdTest.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+    @Test
+    void testFlashcardSetsByAssessmentIds(GraphQlTester graphQlTester) {
+        // Create test data
+        UUID setId1 = UUID.randomUUID();
+        UUID setId2 = UUID.randomUUID();
+        UUID setId3 = UUID.randomUUID();
+
+        UUID assessmentId1 = UUID.randomUUID();
+        UUID assessmentId2 = UUID.randomUUID();
```
Comment (PaulBredl) [inline src/test/java/de/unistuttgart/iste/gits/flashcard_service/api/mutation/query/QueryByAssessmentIdTest.java:45]: It would be more significant test, if you create an assessmentId3 and assign it to flashcardset3. (but keeping the rest as it is, then the expected result is only set1 and set2

File: src/test/java/de/unistuttgart/iste/gits/flashcard_service/api/mutation/MutationDeleteFlashcardSetTest.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    private FlashcardSetRepository flashcardSetRepository;
+
+    @Test
+    @Transactional
+    @Commit
+    void testDeleteFlashcard(GraphQlTester graphQlTester) {
+        // Create and save the flashcard to be deleted
+        FlashcardSetEntity flashcardSet = new FlashcardSetEntity();
+        flashcardSet.setAssessmentId(UUID.randomUUID());
+
```
Comment (PaulBredl) [inline src/test/java/de/unistuttgart/iste/gits/flashcard_service/api/mutation/MutationDeleteFlashcardSetTest.java:43]: Maybe add a flashcard to this set to check if it is deleted as well (assert that no flashcards exists at the end)

File: src/test/java/de/unistuttgart/iste/gits/flashcard_service/api/mutation/MutationDeleteFlashcardTest.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    @Autowired
+    private FlashcardRepository flashcardRepository;
+
+    @Test
+    @Transactional
+    @Commit
+    void testDeleteFlashcard(GraphQlTester graphQlTester) {
+        // Create and save the flashcard to be deleted
+        FlashcardEntity flashcard = new FlashcardEntity();
+        //flashcard.setId(UUID.randomUUID());
```
Comment (PaulBredl) [inline src/test/java/de/unistuttgart/iste/gits/flashcard_service/api/mutation/MutationDeleteFlashcardTest.java]: remove comment

File: src/test/java/de/unistuttgart/iste/gits/flashcard_service/api/mutation/query/QueryByAssessmentIdTest.java [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+
+@Component
```
Comment (PaulBredl) [inline src/test/java/de/unistuttgart/iste/gits/flashcard_service/api/mutation/query/QueryByAssessmentIdTest.java]: Why @Component? not necessary in my oppinion

**Review by PaulBredl — CHANGES_REQUESTED** [review]
Looks really good already, I only have a few suggestions

**Comment by PaulBredl** [comment]
Btw, "QueryByAssessmentIdTest > testFlashcardSetsByAssessmentIds(GraphQlTester) FAILED"
Please fix that

File: src/test/java/de/unistuttgart/iste/gits/flashcard_service/api/mutation/MutationDeleteFlashcardSetTest.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        List<FlashcardSideEntity> sides = new ArrayList<>();
+
+        // Create the first side and set its properties
+        FlashcardSideEntity side1 = new FlashcardSideEntity();
+        side1.setLabel("Side 1");
+        side1.setQuestion(true);
+        side1.setText("Question 1");
+        side1.setFlashcard(flashcard);
+        //flashcardSideRepository.save(side1); // Save the side using the side repository
```
Comment (PaulBredl) [inline src/test/java/de/unistuttgart/iste/gits/flashcard_service/api/mutation/MutationDeleteFlashcardSetTest.java]:
```suggestion
```

File: src/test/java/de/unistuttgart/iste/gits/flashcard_service/api/mutation/MutationDeleteFlashcardSetTest.java [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+        sides.add(side1);
+
+
+        // Create the second side and set its properties
+        FlashcardSideEntity side2 = new FlashcardSideEntity();
+        side2.setLabel("Side 2");
+        side2.setQuestion(false);
+        side2.setText("Answer 1");
+        side2.setFlashcard(flashcard);
+        //flashcardSideRepository.save(side2); // Save the side using the side repository
```
Comment (PaulBredl) [inline src/test/java/de/unistuttgart/iste/gits/flashcard_service/api/mutation/MutationDeleteFlashcardSetTest.java]:
```suggestion
```

**Review by PaulBredl — APPROVED** [review]
